packages/jss/src/dom-monkeypatch.d.ts

```diff
@@ -0,0 +1,638 @@
+/**
+ * Problem: jss doesn't depend on browser env but the types do.
+ * Workaround: Copy/paste subset of interfaces from `dom.d.ts` (TS v3.7.2):
+ * - `CSSStyleRule`
+ * - `StyleSheet`
+ * - `CSSRule`
+ *
+ * Note: Typescript interfaces nicely merge when full dom types are present, but `declare var` doesn't.
+ */
+
+/** A single CSS rule. There are several types of rules, listed in the Type constants section below. */
+interface CSSRule {
+  cssText: string
+  readonly parentRule: CSSRule | null
+  readonly parentStyleSheet: CSSStyleSheet | null
+  readonly type: number
+  readonly CHARSET_RULE: number
+  readonly FONT_FACE_RULE: number
+  readonly IMPORT_RULE: number
+  readonly KEYFRAMES_RULE: number
+  readonly KEYFRAME_RULE: number
+  readonly MEDIA_RULE: number
+  readonly NAMESPACE_RULE: number
+  readonly PAGE_RULE: number
+  readonly STYLE_RULE: number
+  readonly SUPPORTS_RULE: number
+  readonly UNKNOWN_RULE: number
+  readonly VIEWPORT_RULE: number
+}
+
+/** A CSSRuleList is an (indirect-modify only) array-like object containing an ordered collection of CSSRule objects. */
+interface CSSRuleList {
+  readonly length: number
+  item(index: number): CSSRule | null
+  // [index: number]: CSSRule  // Index signatures does not merge nicely.
+}
+
+/** An object that is a CSS declaration block, and exposes style information and various style-related methods and properties. */
+interface CSSStyleDeclaration {
```

Comment:
Is this for server-side rendering? Is react-dom doing that as well? I think ultimately this creates a maintenance burden because now we will have to maintain those types.

We def. need a way to avoid this. If anything types from styles should be generated as from mdn-data package and we have a separate issue for this.